### ISSUE TITLE
Failed to Attach network endpoint(s) error

### ISSUE THREAD

**Issue by gabihodoroaga** [kubernetes/ingress-gce#1463]
Hi,

I did some tests related ot the Ingress, I setup my services with ```cloud.google.com/neg: '{"ingress": true}'``` annotation but the load balancer was not able to connect to the backend pods.

I checked and  NEG is was created but there are not endpoints attached to it.
And then I found this error in logs:

> Failed to Attach 2 network endpoint(s) (NEG "k8s1-c5d09f7f-default-nginx-service-80-c3aae96e" in zone "us-central1-    a"): googleapi: Error 400: Invalid value for field 'resource.ipAddress': '10.0.0.13'. Specified IP address 10.0.0.13 doesn't belong to the (sub)network default or to the instance gke-demo-gce-cluster-default-pool-18343f80-4cnf., invalid

After many hours of debugging I found that the issue is that the cluster was not created with "VPC-native routing" enabled (aka --enable-ip-alias, UseIpAlias).

When "VPC-native routing" is enabled the pods subnet is added to the VPC subnets of the zone where the cluster was created, and the pods will have IPs allocated from the VPC subnet. This is how the magic is happening and why it is possible to access the pods directly from the LB.

Basically NEG is not supported when the "VPC-native routing" is not enabled.

I would be nice to **NOT** enable NEG for the services, even if they have the neg annotation, and raise a warning in logs for the user and explain why NEG cannot be used (not supported by the cluster).

Thanks
gabihodoroaga
